homeassistant/components/vallox/__init__.py

```diff
@@ -291,3 +316,37 @@ async def async_handle(self, call: ServiceCall) -> None:
         # be observed by all parties involved.
         if result:
             await self._coordinator.async_request_refresh()
+
+
+class ValloxEntity(CoordinatorEntity[ValloxDataUpdateCoordinator]):
+    """Representation of a Vallox entity."""
+
+    _device_config_url: str | None
+
+    def __init__(self, name: str, coordinator: ValloxDataUpdateCoordinator) -> None:
+        """Initialize a Vallox entity."""
+        super().__init__(coordinator)
+
+        if self.coordinator.config_entry is not None:
+            self._device_config_url = (
+                f"http://{self.coordinator.config_entry.data[CONF_HOST]}"
+            )
+        else:
+            self._device_config_url = None
+
+        self._device_uuid = self.coordinator.data.get_uuid()
+        self._device_model = self.coordinator.data.get_model()
+        self._device_name = name
+        self._device_sw_version = self.coordinator.data.get_sw_version()
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device information of the entity."""
+        return DeviceInfo(
```

Comment:
Set this as `self._attr_device_info` instead.

You can add the `configuration_url` later instead of setting it to `None`

Reply:
Could you elaborate please? I think when I wrote this, the thing with config_url being None was just to dance around mypy.

Reply:
@bdraco was it something like this you had in mind? Seems to work locally for me at least, no complaints from mypy, and tests are passing.

```diff
diff --git homeassistant/components/vallox/__init__.py homeassistant/components/vallox/__init__.py
index e7d916359f..8b12315d8e 100644
--- homeassistant/components/vallox/__init__.py
+++ homeassistant/components/vallox/__init__.py
@@ -321,32 +321,26 @@ class ValloxServiceHandler:
 class ValloxEntity(CoordinatorEntity[ValloxDataUpdateCoordinator]):
     """Representation of a Vallox entity."""
 
-    _device_config_url: str | None
+    _attr_device_info: DeviceInfo
 
     def __init__(self, name: str, coordinator: ValloxDataUpdateCoordinator) -> None:
         """Initialize a Vallox entity."""
         super().__init__(coordinator)
 
-        if self.coordinator.config_entry is not None:
-            self._device_config_url = (
-                f"http://{self.coordinator.config_entry.data[CONF_HOST]}"
-            )
-        else:
-            self._device_config_url = None
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(self.coordinator.data.get_uuid()))},
+            manufacturer=DEFAULT_NAME,
+            model=self.coordinator.data.get_model(),
+            name=name,
+            sw_version=self.coordinator.data.get_sw_version(),
+        )
 
-        self._device_uuid = self.coordinator.data.get_uuid()
-        self._device_model = self.coordinator.data.get_model()
-        self._device_name = name
-        self._device_sw_version = self.coordinator.data.get_sw_version()
+        if self.coordinator.config_entry is not None:
+            self._attr_device_info[
+                "configuration_url"
+            ] = f"http://{self.coordinator.config_entry.data[CONF_HOST]}"
 
     @property
     def device_info(self) -> DeviceInfo:
         """Return the device information of the entity."""
-        return DeviceInfo(
-            configuration_url=self._device_config_url,
-            identifiers={(DOMAIN, str(self._device_uuid))},
-            manufacturer=DEFAULT_NAME,
-            model=self._device_model,
-            name=self._device_name,
-            sw_version=self._device_sw_version,
-        )
+        return self._attr_device_info

```

Reply:
You can drop
```
 @property
     def device_info(self) -> DeviceInfo:
```

and
```
    _attr_device_info: DeviceInfo
```

Reply:
@slovdahl could you send a PR against my branch?

Reply:
> @slovdahl could you send a PR against my branch?

Sure 👍 https://github.com/andre-richter/core/pull/1

Reply:
Thanks, merged 👍

Reply:
@bdraco is this resolved now?